utils/s3/client.cc

```diff
@@ -979,8 +979,8 @@ class client::do_upload_file {
         req.query_parameters.emplace("partNumber", to_sstring(part_number + 1));
         req.query_parameters.emplace("uploadId", _upload_id);
         s3l.trace("PUT part {}, {} bytes (upload id {})", part_number, part_size, _upload_id);
-        req.write_body("bin", part_size, [f=std::move(f), mem_units=std::move(mem_units), offset, part_size] (output_stream<char>&& out_) mutable {
-            auto input = make_file_input_stream(std::move(f), offset, part_size, input_stream_options());
+        req.write_body("bin", part_size, [f=std::move(f), mem_units=std::move(mem_units), offset, part_size] (output_stream<char>&& out_) {
+            auto input = make_file_input_stream(f, offset, part_size, input_stream_options());
```

Comment:
Is there a test covering this? Sounds like this causes use-after-move when hit.

Reply:
probably not. we would need a dedicated fixture for mocking this temporary failure.

Reply:
No, there is no test, @kreuzerkrieg is working it.
And yes, it's use-after-move, file object when moved-from points to nullpre impl and doesn't check it

Reply:
> Sounds like this causes use-after-move when hit.

Still no need to backport, this uploader code appeared after 6.1